opentelemetry-sdk/tests/metrics/test_instrument.py

```diff
@@ -324,6 +325,30 @@ def test_record(self):
         hist.record(1.0)
         mc.consume_measurement.assert_called_once()
 
+    def test_time(self):
+        mc = Mock()
+        hist = Histogram("name", Mock(), mc)
+
+        @hist.time()
```

Comment:
I think I wasn't clear, sorry, let me try again:

In the current implementation, if you use the "timing histogram" object as a context manager, there **is a** way of associating attributes to the measurement (by calling `set_attributes` on the context manager object).

How would you associate attributes to the measurement when using the "timing histogram" object as a decorator? Like this?

``` python
@timing_histogram.time()
def function():
    sleep(1)
    timing_histogram.set_attributes({"a": "b"})
```

That would make the attributes to be set every time the function is called which is not ideal. Also, the attributes would be "hardcoded" in the function which is not the same that can be achieved with a context manager where the attributes can change:

``` python
for i in range(3):
    with timing_histogram.time() as timing_histogram_measurer:
        sleep(1)
        timing_histogram_measurer.set_attributes({"a": str(i)})
````

I think this functionality should be equivalent. Now, a question, which behavior do we want? Does it make sense/is useful to allow the user to associate different attributes to the time measurement every time the measurement is done?

Reply:
Every time the time is recorded the attributes are reset and if they are setting again I imagine they would expect it to present.